File: app/src/main/java/com/absinthe/kage/device/cmd/PromptPhoneConnectedCommand.java
package com.absinthe.kage.device.cmd;

import com.absinthe.kage.connect.protocol.IpMessageConst;
import com.absinthe.kage.connect.protocol.IpMessageProtocol;
import com.absinthe.kage.device.Command;
import com.absinthe.kage.device.CommandBuilder;
import com.absinthe.kage.device.client.Client;
import com.absinthe.kage.device.model.DeviceInfo;

public class PromptPhoneConnectedCommand extends Command {

    public static final int LENGTH = 3;

    public String phoneName;
    public String localIp;

    public PromptPhoneConnectedCommand() {
        cmd = IpMessageConst.PROMPT_PHONE_CONNECT;
    }

    @Override
    public String pack() {
        return new CommandBuilder()
                .with(this)
                .append(phoneName)
                .append(localIp)
                .build();
    }

    @Override
    public void doWork(Client client, String received) {
        if (client.getDeviceInfo() == null) {
            client.setDeviceInfo(new DeviceInfo());
        }
        if (parseReceived(received)) {
            client.getDeviceInfo().setName(phoneName);
            client.getDeviceInfo().setIp(localIp);
        }
    }

    @Override
    public boolean parseReceived(String received) {
        String[] splits = received.split(IpMessageProtocol.DELIMITER);
        if (splits.length == LENGTH) {
            phoneName = splits[1];
            localIp = splits[2];
            return true;
        } else {
            return false;
        }
    }
}

File: app/src/main/kotlin/com/absinthe/kage/device/cmd/PromptPhoneConnectedCommand.kt
package com.absinthe.kage.device.cmd

import com.absinthe.kage.connect.protocol.IpMessageConst
import com.absinthe.kage.connect.protocol.IpMessageProtocol
import com.absinthe.kage.device.Command
import com.absinthe.kage.device.CommandBuilder
import com.absinthe.kage.device.client.Client
import com.absinthe.kage.device.model.DeviceInfo

class PromptPhoneConnectedCommand : Command() {

    var phoneName: String? = null
    var localIp: String? = null

    override fun pack(): String {
        return CommandBuilder()
                .with(this)
                .append(phoneName)
                .append(localIp)
                .build()
    }

    override fun doWork(client: Client, received: String) {
        if (client.deviceInfo == null) {
            client.deviceInfo = DeviceInfo()
        }
        if (parseReceived(received)) {
            client.deviceInfo!!.name = phoneName!!
            client.deviceInfo!!.ip = localIp!!
        }
    }

    override fun parseReceived(received: String): Boolean {
        val splits = received.split(IpMessageProtocol.DELIMITER).toTypedArray()

        return if (splits.size == LENGTH) {
            phoneName = splits[1]
            localIp = splits[2]
            true
        } else {
            false
        }
    }

    companion object {
        const val LENGTH = 3
    }

    init {
        cmd = IpMessageConst.PROMPT_PHONE_CONNECT
    }
}